### PR TITLE
Scale TES to be able to run reasonably-sized workflows on Funnel on Kubernetes with the AWS job store

### PR DESCRIPTION
**Pull request by adamnovak**
This is where I'm collecting the changes I need to actually use TES.

I think I am going to need to pass in the AWS credentials via environment variables; I'm going to just suck them out of Kubernetes before I start my TES workflow so they work without MFA.

Major changes so far include how I forgot to pass any requirements along to TES, and how we have to deal with TES tasks not having a writable container root filesystem. I'm also making sure to fish out the logs for failed TES tasks, like we do for failed Kubernetes pods.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

* TES batch system now should be able to run workflows on Funnel on Kubernetes

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

**Comment by adamnovak** [comment]
I'm going to stop working on this for now. I think all the fixes are necessary, but I'm not going to assert they're sufficient for getting TES workflows really working well on Funnel Kubernetes. It's going to have part of #3927, but probably not all of it.

File: src/toil/__init__.py [hunk 1]
```diff
@@ -637,8 +637,18 @@ def _obtain_credentials_from_cache_or_boto3(self):
                         dir_path = os.path.dirname(path)
                         if not os.path.exists(dir_path):
                             log.debug('Creating parent directory %s', dir_path)
-                            # A race would be ok at this point
-                            os.makedirs(dir_path, exist_ok=True)
+                            try:
+                                # A race would be ok at this point
+                                os.makedirs(dir_path, exist_ok=True)
+                            except OSError as e2:
+                                if e.errno == errno.EROFS:
```
Comment (DailyDreaming) [inline src/toil/__init__.py:644]: Should this be `e2.errno`?

**Review by DailyDreaming — CHANGES_REQUESTED** [review]

File: src/toil/__init__.py (hunk 1 above)
Comment (adamnovak) [reply src/toil/__init__.py:644]: Yes it should!

**Review by DailyDreaming — APPROVED** [review]
LGTM.

**Pull request merged**